src/plugin.ts

```diff
@@ -438,7 +438,7 @@ export function dtsPlugin(options: PluginOptions = {}): import('vite').Plugin {
         const typesPath = types ? resolve(root, types) : resolve(outputDir, indexName)
 
         for (const name of entryNames) {
-          let filePath = multiple ? resolve(outputDir, name.replace(tsRE, '.d.ts')) : typesPath
+          let filePath = multiple ? resolve(outputDir, `${name}.d.ts`) : typesPath
```

Comment:
We should judge whether the extension is included here, because it will fail when extension exists (`xxx.ts` -> `xxx.ts.d.ts`).

Reply:
And we should apply the same change to line 491.

Reply:
Makes sense 👍 

Also, I'm no longer using your library and don't have time to contribute anymore. Feel free to pick up this PR where I left off.

Reply:
All right, thanks for you contributions, I will fix it directly.